kong/router/fields.lua

```diff
@@ -298,6 +299,13 @@ if is_http then
         -- http.path.segments.0 => params.segments[1 + 0] => a
         -- http.path.segments.1_2 => b/c
 
+        if range == "len" then
+          value = #segments
+          segments[range] = value
+
+          return value
+        end
+
```

Comment:
We can put `http.path.segments.len` directly into `FIELDS_FUNCS`, this should override the `http.path.segments.*` function.

Reply:
I prefer to implement here, because we will split the uri, or else we have to repeat the split code.

Reply:
Can you do a `get_segments(param)` instead? We can simply do a refactor to load and cache the segments. Logically it is clearer than calling `len` the range.

Reply:
done